### ISSUE TITLE
📝 arrow function with long type and single param without parens formats to invalid code

### ISSUE THREAD

**Issue by arilotter** [biomejs/biome#1905]
### Environment information

```bash
CLI:
  Version:                      1.4.1
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v21.5.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.10.5"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              true
  Organize imports disabled:    true
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Configuration

```JSON
{
  "$schema": "./node_modules/@biomejs/biome/configuration_schema.json",
  "javascript": {
    "formatter": {
      "arrowParentheses": "asNeeded"
    }
  }
}
```

### Issue information
Biome outputs invalid TS; the `=>` is on a new line, that's not valid.
```ts
type InstanceID = string;
type MaybeCardWithAttachment = string;
function outerFunctionToForceIndent() {
	const cardWithAttachment: (id: InstanceID) => MaybeCardWithAttachment =
		id
	=> {
		return `${id}test`;
	};
}
```

### Playground link

[https://codesandbox.io/p/devbox/biome-starter-cbs-rky6zq](https://codesandbox.io/p/devbox/biome-starter-cbs-rky6zq?file=%2Fsrc%2Findex.js%3A9%2C2&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clsz38eke00052069kcfismhs%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clsz38eke000220696t2erae9%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clsz38eke00032069bacix6yv%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clsz38eke00042069649ybzce%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B100%252C0%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clsz38eke000220696t2erae9%2522%253A%257B%2522id%2522%253A%2522clsz38eke000220696t2erae9%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsz38eke000120695hb7s1f1%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252F.codesandbox%252FDockerfile%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clsz38eke000120695hb7s1f1%2522%257D%252C%2522clsz38eke00042069649ybzce%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clsz38eke00042069649ybzce%2522%257D%252C%2522clsz38eke00032069bacix6yv%2522%253A%257B%2522id%2522%253A%2522clsz38eke00032069bacix6yv%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsz3ceul00qc206axydq4ypd%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clsz3cewd001qdkip4i2b83tc%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clsz3ceul00qc206axydq4ypd%2522%257D%257D%252C%2522showDevtools%2522%253Afalse%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by togami2864** [comment]
biome playground: https://biomejs.dev/playground/?lineWidth=140&arrowParentheses=as-needed&unsafeParameterDecoratorsEnabled=false&allowComments=false&code=dAB5AHAAZQAgAEkAbgBzAHQAYQBuAGMAZQBJAEQAIAA9ACAAcwB0AHIAaQBuAGcAOwAKAHQAeQBwAGUAIABNAGEAeQBiAGUAQwBhAHIAZABXAGkAdABoAEEAdAB0AGEAYwBoAG0AZQBuAHQAIAA9ACAAcwB0AHIAaQBuAGcAOwAKAGYAdQBuAGMAdABpAG8AbgAgAG8AdQB0AGUAcgBGAHUAbgBjAHQAaQBvAG4AVABvAEYAbwByAGMAZQBJAG4AZABlAG4AdAAoACkAIAB7AAoACQBjAG8AbgBzAHQAIABjAGEAcgBkAFcAaQB0AGgAQQB0AHQAYQBjAGgAbQBlAG4AdAA6ACAAKABpAGQAOgAgAEkAbgBzAHQAYQBuAGMAZQBJAEQAKQAgAD0APgAgAE0AYQB5AGIAZQBDAGEAcgBkAFcAaQB0AGgAQQB0AHQAYQBjAGgAbQBlAG4AdAAgAD0ACgAJAAkAaQBkAAoACQA9AD4AIAB7AAoACQAJAHIAZQB0AHUAcgBuACAAYAAkAHsAaQBkAH0AdABlAHMAdABgADsACgAJAH0AOwAKAH0A&jsx=false

**Comment by togami2864** [comment]
The error occurs due to a line break before the arrow. [ref](https://github.com/microsoft/TypeScript/issues/2282)

We have two options:

- Treat it as a special case, accept the error, and format accordingly ([as Prettier does](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEMCeAHOACAklAZxgEMow5cARbAXmyICcBLKAcwG4AdWTHAWWJoARnADCxBgBMA6kxgALAIIwSYeQFsEMWvRjM2XKADMArmRhNo2CCfgMAYmbAXoAFQj2IDcvklaAFACU2MDcnDCQhNpgEjJySirEapqwSNj+TJJp+ESkPpTBNAB82ALCYrGyCsqqGlq0YTDhmY3FIY3hDHAwJgxQ2AAGACTAmQC+8EQDhuFjhmMgADQgEBguhMigEgwQAO4AChIIBMggxAA2u4Iny0IMSQDW3QDKxJoAMixwyEYXBHC3e5gJ4wZ4YJIsVjIPQmAEgf7qJjQhiw5ZwAAeWGYKRI51ccD6EiYcBOSF+53+ywIkPOcAAiiYIPAfn84QArAjo540+mM5lk1nLACOfLg+x2GFJZwIAFooHA4H5JEtUPcmOdIaIIOp1MRThdzirqWxaTVmEJbGKCZ95SyKXD5DB1OdpPI5CTweRnsc5EwAG5yNCnMAEG4gP2w3xaZ5gZhrRRQSTPdC0u2UkAYHb-aT3DCnTMkgl+77LFj-BgwcXEVi6tNw8EMcundBYAixphrFWZlgwWSSBTIAAcAAZll0RUwulWa3qBfbliQhH2B0gAEzLEz-VzEISk8npuDqESSJXvUisEzVuCeBi6lSQ-W2CAgMZjIA). I'm don't know why this error occurs in the Prettier section of the Biome playground 🤔).
- Treat it as invalid code ([as TypeScript and Biome does](https://www.typescriptlang.org/play?#code/C4TwDgpgBAkgdgZ2AQzgYwjAIlAvFJAJwEs4BzAbgChRIoBZZEAIwgGFlCATAdWOAAWAQWAo0AgLYQ4wPAWAly1AGYBXdMGIB7OFC2rgEQgDF1aTToAqW41sIZ4XacAAUASigBvKgEg0OpCg0Tl5+YVFkcSkZAC4oF2IuOPgkVAcsD1wAPgYmVg5uPkERMUlnPF8fRN9sr0rCCGBVQl0AAwAST0SAX0MkVuofbupuoA)).

Should we adjust our code to be compatible with Prettier🤔

**Comment by Sec-ant** [comment]
I think what OP trying to say is the valid code like below will be formatted to invalid code? 

```ts
type InstanceID = string;
type MaybeCardWithAttachment = string;
function outerFunctionToForceIndent() {
	const cardWithAttachment: (id: InstanceID) => MaybeCardWithAttachment = (
		id
	) => {
		return `${id}test`;
	};
}
```

![image](https://github.com/biomejs/biome/assets/10386119/4678df1a-5e7c-4743-b1cb-7af58399fa5c)


[Playground Link](https://biomejs.dev/playground/?arrowParentheses=as-needed&code=dAB5AHAAZQAgAEkAbgBzAHQAYQBuAGMAZQBJAEQAIAA9ACAAcwB0AHIAaQBuAGcAOwAKAHQAeQBwAGUAIABNAGEAeQBiAGUAQwBhAHIAZABXAGkAdABoAEEAdAB0AGEAYwBoAG0AZQBuAHQAIAA9ACAAcwB0AHIAaQBuAGcAOwAKAGYAdQBuAGMAdABpAG8AbgAgAG8AdQB0AGUAcgBGAHUAbgBjAHQAaQBvAG4AVABvAEYAbwByAGMAZQBJAG4AZABlAG4AdAAoACkAIAB7AAoACQBjAG8AbgBzAHQAIABjAGEAcgBkAFcAaQB0AGgAQQB0AHQAYQBjAGgAbQBlAG4AdAA6ACAAKABpAGQAOgAgAEkAbgBzAHQAYQBuAGMAZQBJAEQAKQAgAD0APgAgAE0AYQB5AGIAZQBDAGEAcgBkAFcAaQB0AGgAQQB0AHQAYQBjAGgAbQBlAG4AdAAgAD0AIAAoAAoACQAJAGkAZAAKAAkAKQAgAD0APgAgAHsACgAJAAkAcgBlAHQAdQByAG4AIABgACQAewBpAGQAfQB0AGUAcwB0AGAAOwAKAAkAfQA7AAoAfQA%3D&jsx=false).

**Comment by togami2864** [comment]
@Sec-ant  Nice catch! Maybe I misunderstood.
@arilotter Could you provide your input? The playground link you attached seems empty.

**Comment by arilotter** [comment]
> @Sec-ant  Nice catch! Maybe I misunderstood.
> @arilotter Could you provide your input? The playground link you attached seems empty.

sec-ant has it nailed! biome formats the code into TS that doesn't compile, because it's illegal to have `=>` on its own line.
